### PR TITLE
internal/exec/resolvable: include struct field name in errors

### PR DESCRIPTION
**Pull request by agnivade**
We were only adding method name, which meant that it was taking
an empty string if the resolver was a struct field. This was
making the error messages hard to parse as the user can't know
which field has the error.

Added a check to use the correct variable.

Earlier:

```
can not use int64 as Float
        used by (*api4.channelMember).
```

Now:

```
can not use int64 as Float
        used by (*api4.channelMember).LastViewedAt
```

### CODE REVIEW

File: graphql_test.go [hunk 1]
```diff
@@ -3217,14 +3226,29 @@ func TestInputArguments_failSchemaParsing(t *testing.T) {
...
+		"Struct field name inclusion": {
+			Args: args{
+				Resolver: &helloOutput{},
+				Opts:     []graphql.SchemaOpt{graphql.UseFieldResolvers()},
+				Schema: `
+					type Query {
+						hello(): HelloOutput!
+					}
+					type HelloOutput {
+						name: Int
```
Comment (pavelnikolov) [inline graphql_test.go:3238]: Why are you using an `Int` here when the type of the `Name` field is `string`? I do not understand the problem that you are trying to solve.

File: graphql_test.go [hunk 2]
```diff
@@ -3217,14 +3226,29 @@ func TestInputArguments_failSchemaParsing(t *testing.T) {
 			},
 			Want: want{Error: "field \"Input\": *struct { Thing string } does not define field \"name\" (hint: missing `args struct { ... }` wrapper for field arguments, or missing field on input struct)\n\tused by (*graphql_test.inputArgumentsObjectMismatch3).Hello"},
 		},
+		"Struct field name inclusion": {
+			Args: args{
+				Resolver: &helloOutput{},
```
Comment (pavelnikolov) [inline graphql_test.go:3231]: According to your schema you need to have a field or a method named `Hello`, which returns an instance of `helloOutput`. Instead you are using the `helloOutput` instance as a resolver. This doesn't make sense to me.

File: internal/exec/resolvable/resolvable.go [hunk 3]
```diff
@@ -256,7 +256,13 @@ func (b *execBuilder) makeObjectExec(typeName string, fields types.FieldsDefinit
...
 		fe, err := b.makeFieldExec(typeName, f, m, sf, methodIndex, fieldIndex, methodHasReceiver)
 		if err != nil {
-			return nil, fmt.Errorf("%s\n\tused by (%s).%s", err, resolverType, m.Name)
+			var resolverName string
+			if methodIndex != -1 {
+				resolverName = m.Name
+			} else {
+				resolverName = sf.Name
+			}
+			return nil, fmt.Errorf("%s\n\tused by (%s).%s", err, resolverType, resolverName)
```
Comment (pavelnikolov) [inline internal/exec/resolvable/resolvable.go:265]: Can you give me a good example before and after this change so that I can understand what exactly you are trying to achieve.

File: graphql_test.go [hunk 4]
```diff
@@ -3078,6 +3078,14 @@ type helloInput struct {
 	Name string
 }
 
+type helloOutput struct {
+	Name string
+}
+
+func (*helloOutput) Hello() helloOutput {
+	return helloOutput{}
+}
```
Comment (pavelnikolov) [inline graphql_test.go:3087]: Why would the `helloOutput` have a `Hello()` method that returns a new instance of `helloOutput`?

File: internal/exec/resolvable/resolvable.go (hunk 3 above)
Comment (pavelnikolov) [reply internal/exec/resolvable/resolvable.go:265]: In what case the error message would be ambiguous?

File: graphql_test.go (hunk 2 above)
Comment (agnivade) [reply graphql_test.go:3231]: Woops, messed up that while writing the test. Fixing it.

File: graphql_test.go (hunk 1 above)
Comment (agnivade) [reply graphql_test.go:3238]: I tried to explain the problem clearly in the PR message, but let me explain with an example:

Currently, when you have a type mismatch, the error message will say something like:

```
 can not use int64 as Float
        used by (*api4.channelMember).
```

Notice that there is nothing after the `.`? That is because when resolving struct methods, we don't print the field name. We only print `m.Name` always instead of `m.Name` or `sf.Name` depending on the condition. That's what this PR fixes.

Now, with the fix, it will print as:
```
can not use int64 as Float
        used by (*api4.channelMember).LastViewedAt
```

Does that help?

File: internal/exec/resolvable/resolvable.go (hunk 3 above)
Comment (agnivade) [reply internal/exec/resolvable/resolvable.go:265]: Explained above and also added in the PR message.

**Comment by agnivade** [comment]
@pavelnikolov - Thanks for the review! The test was my bad. I pushed with a separate struct. PTAL.
